### PR TITLE
Require Client Certificates

### PR DESCRIPTION
**Pull request by jaraco**
This is a re-submission of #52.

---
This change is [Reviewable](https://reviewable.io/reviews/cherrypy/cheroot/108)

### CODE REVIEW

**Comment by webknjaz** [comment]
Why didn't you just push to original branch?

**Comment by webknjaz** [comment]
This needs rebase in order to bypass checks

**Comment by jaraco** [comment]
> Why didn't you just push to original branch?

I tried, but I didn't have permission.

**Comment by jaraco** [comment]
> This needs rebase in order to bypass checks

The PR handles this. The commits themselves fail tests, but the PR build performs the merge and runs the tests on that, which are passing (https://travis-ci.org/cherrypy/cheroot/builds/422106587).

File: cheroot/test/test_https.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+# vim: set fileencoding=utf-8 :
+
+import os
+import pytest
+import ssl
+
+from cheroot.ssl.builtin import BuiltinSSLAdapter
+from cheroot.test import helper
+import cheroot
+import ddt
```
Comment (webknjaz) [inline cheroot/test/test_https.py]: https://github.com/cherrypy/cheroot/pull/52/files#r213702146

**Pull request merged**